src/main/java/dao/Dao.java

```diff
@@ -0,0 +1,18 @@
+package dao;
+
+import java.util.Optional;
+
+public interface Dao<T> {
```

Comment:
Do you really need to use generic Dao?  You have only one Dao implementation. The method arguments names related to cards that make the interface more narrow in using.

(BTW I like that you know about it)

Reply:
removed this interface